lib/model/model.go

```diff
@@ -2554,6 +2567,70 @@ func (m *model) CommitConfiguration(from, to config.Configuration) bool {
 	return true
 }
 
+func (m *model) cleanPending(cfg config.Configuration, removedFolders map[string]bool) {
+	// Cache to maps for easy lookups
+	ignoredDevices := make(map[protocol.DeviceID]bool, len(cfg.IgnoredDevices))
+	for _, dev := range cfg.IgnoredDevices {
+		ignoredDevices[dev.ID] = true
+	}
+	var existingDevices map[protocol.DeviceID]config.DeviceConfiguration
+	existingDevices = cfg.DeviceMap()
+	existingFolders := make(map[string]*config.FolderConfiguration, len(cfg.Folders))
+	for i := range cfg.Folders {
+		folder := &cfg.Folders[i]
+		existingFolders[folder.ID] = folder
+	}
+
+	pendingFolder, err := m.db.NewPendingFolderIterator(nil)
```

Comment:
This iterator is never released, illustrating that the API is needlessly hard to use correctly.

Reply:
Yikes, just as I said it's used everywhere... Thanks for catching, now fixed.